keras_cv/layers/object_detection/nms_prediction_decoder.py

```diff
@@ -15,115 +15,98 @@
 import tensorflow as tf
 
 from keras_cv import bounding_box
-from keras_cv import layers as cv_layers
 
 
-class NmsPredictionDecoder(tf.keras.layers.Layer):
+# TODO(tanzhenyu): provide a TPU compatible NMS decoder.
+class NmsDecoder(tf.keras.layers.Layer):
     """A Keras layer that decodes predictions of an object detection model.
 
-    By default, NmsPredictionDecoder uses a
-    `keras_cv.layers.NonMaxSuppression` layer to perform box pruning.  The layer may
-    optionally take a `suppression_layer`, which can perform an alternative suppression
-    operation, such as SoftNonMaxSuppression.
-
     Arguments:
-      classes: Number of classes in the dataset.
       bounding_box_format: The format of bounding boxes of input dataset. Refer
         [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
         for more details on supported bounding box formats.
-      anchor_generator: a `keras_cv.layers.AnchorGenerator`.
-      suppression_layer: (Optional) a `keras.layers.Layer` that follows the same API
-        signature of the `keras_cv.layers.NonMaxSuppression` layer.  This layer should
-        perform a suppression operation such as NonMaxSuppression, or
-        SoftNonMaxSuppression.
-      box_variance: (Optional) The scaling factors used to scale the bounding box
-        targets.  Defaults to `(0.1, 0.1, 0.2, 0.2)`.  **Important Note:**
-        `box_variance` is applied to the boxes in `xywh` format.
+      from_logits: boolean, True means input score is logits, False means confidence.
+      iou_threshold: a float value in the range [0, 1] representing the minimum
+        IoU threshold for two boxes to be considered same for suppression. Defaults
+        to 0.5.
+      confidence_threshold: a float value in the range [0, 1]. All boxes with
+        confidence below this value will be discarded. Defaults to 0.05.
+      max_detections: the maximum detections to consider after nms is applied. A large
+        number may trigger significant memory overhead. Defaults to 100.
+      max_detections_per_class: the maximum detections to consider per class after
+        nms is applied. Defaults to 100.
     """
 
     def __init__(
         self,
         bounding_box_format,
-        anchor_generator,
-        classes=None,
-        suppression_layer=None,
-        box_variance=(0.1, 0.1, 0.2, 0.2),
+        from_logits,
+        iou_threshold=0.5,
+        confidence_threshold=0.5,
+        max_detections=10,
```

Comment:
These defaults don't align with the docstrings, which say that defaults are 100 for each

Reply:
Good catch. Done